File: TP1py/Produk.py
# Class Produk untuk menyimpan data barang elektronik
class Produk:
    # Constructor
    def __init__(self, id="", nama="", merek="", kategori="", harga=0, stok=0):
        # atribut private
        self.__id = id
        self.__nama = nama
        self.__merek = merek
        self.__kategori = kategori
        self.__harga = harga
        self.__stok = stok
        

    # Getter untuk mengambil nilai atribut private
    def get_id(self):           #  mengambil nilai ID
        return self.__id

    def get_nama(self):         #  mengambil nama
        return self.__nama

    def get_merek(self):        # mengambil merek
        return self.__merek

    def get_kategori(self):     #  mengambil kategori
        return self.__kategori

    def get_harga(self):        #  mengambil nilai harga
        return self.__harga

    def get_stok(self):
        return self.__stok      # mengambil stok


    # Setter untuk mengubah nilai atribut private
    def set_nama(self, nama):           # mengubah nama
        self.__nama = nama

    def set_merek(self, merek):         # mengubah merek
        self.__merek = merek

    def set_kategori(self, kategori):   # mengubah kategori
        self.__kategori = kategori

    def set_harga(self, harga):         # mengubah harga
        self.__harga = harga

    def set_stok(self, stok):           # mengubah stok
        self.__stok = stok

    # Menampilkan semua data produk
    def tampil(self, no):
        print(f"{no:<3} | {self.__id:<8} | {self.__nama:<20} | "
              f"{self.__merek:<15} | {self.__kategori:<12} | "
              f"{self.__harga:>8} | {self.__stok:>5} | ")

File: TP1java/Produk.java
// Class Produk untuk menyimpan data barang elektronik
public class Produk {
    // atribut private agar tidak bisa langsung diakses dari luar
    private String id;       // atribut unik produk
    private String nama;     // nama produk
    private String merek;    // atribut merek
    private String kategori; // kategori produk
    private int harga;       // harga produk
    private int stok;        // atribut stok

    // Constructor
    public Produk(String id, String nama, String merek, String kategori, int harga, int stok) {
        this.id = id;
        this.nama = nama;
        this.merek = merek;
        this.kategori = kategori;
        this.harga = harga;
        this.stok = stok;
    }

    // Setter (untuk mengubah data)
    public void setNama(String nama) { this.nama = nama; }
    public void setMerek(String merek) { this.merek = merek; }
    public void setKategori(String kategori) { this.kategori = kategori; }
    public void setHarga(int harga) { this.harga = harga; }
    public void setStok(int stok) { this.stok = stok; }

    // Getter (untuk membaca data privat)
    public String getId() { return id; }
    public String getNama() { return nama; }
    public String getMerek() { return merek; }
    public String getKategori() { return kategori; }
    public int getHarga() { return harga; }
    public int getStok() { return stok; }

    // Tambah stok
    public void tambahStok(int jumlah) {
        this.stok += jumlah;
    }

    // Menampilkan semua data produk
    public void tampil(int no) {
        System.out.printf("%-3d | %-8s | %-20s | %-15s | %-12s | %10d | %5d |%n",
                no, id, nama, merek, kategori, harga, stok);
    }
}
